daisy_workflows/image_build/install_package/cos/package_replacement/compile_debian_package.sh

```diff
@@ -0,0 +1,167 @@
+  cat /etc/debian_version
+  DEB_VERSION=$(</etc/debian_version)
+  # deb_version=${DEB}
+  if [[ "$DEB_VERSION" =~ "bookworm" ]]; then
+      export DEB="12"
+  else
+      export DEB="11"
+  fi
+}
+
+install_go(){
+  # Install the correct version of Go.
+  echo -e "\nATTENTION: Installing go...\n"
+  cd /workspace
+  source ./install_go.sh; install_go /tmp/go
+}
+
+upstream_tarball() {
+  # Create the build directory and the upstream tarball.
+  cd guest-agent
+  mkdir /tmp/debpackage
+  export BUILD_DIR="/tmp/debpackage"
+  export VERSION="112723"
+  export TAR="${PKGNAME}_${VERSION}.orig.tar.gz"
+  echo -e "\nATTENTION: Creating build dir and upstream tarball...\n"
+  tar czvf "${BUILD_DIR}/${TAR}" --exclude .git --exclude packaging \
+    --transform "s/^\./${PKGNAME}-${VERSION}/" .
+}
+
+build_tarball(){
+  # Extract and build the tarball.
+  echo -e "\nATTENTION: Extracting tarball and building...\n"
+  tar -C "$BUILD_DIR" -xzvf "${BUILD_DIR}/${TAR}"
+  cp -r packaging/debian "${BUILD_DIR}/${PKGNAME}-${VERSION}/"
+  cd "${BUILD_DIR}/${PKGNAME}-${VERSION}"
+  [[ -f debian/changelog ]] && rm debian/changelog
+  export RELEASE="g1${DEB}"
+  dch --create -M -v 1:${VERSION}-${RELEASE} --package $PKGNAME -D stable \
+    "Debian packaging for ${PKGNAME}"
+  debuild -us -uc
+  cd ..
+}
+
+unpack_binaries(){
+  # Unpack the binaries into a local directory that contains 1) the .deb files
+  # (which provide installation paths) and 2) binaries in a localized dir format.
+  echo -e "\nATTENTION: Unpacking binaries in a directory...\n"
+  dpkg-deb -x google-guest-agent_112723-g111_amd64.deb debian_binaries
+  dpkg-deb -c google-guest-agent_112723-g111_amd64.deb >> google-guest-agent-readable-deb.txt
+  echo -e "\nATTENTION: google-guest-agent.deb file contents below...\n"
+  cat google-guest-agent-readable-deb.txt
+  mv debian_binaries /workspace/upload
+  mv google-guest-agent-readable-deb.txt /workspace
+}
+
+identify_replacement_files(){
+  echo -e "\nATTENTION: Creating a list of files to replace...\n"
+  file="/workspace/google-guest-agent-readable-deb.txt"
+  repl_file="/workspace/repl_files.txt"
+
+  # Go through every line in the deb file...
+  while read -r line; do
+
+    # For every line...
+    read -ra arr <<< "$line"
+
+    # Store file permissions and the file path eg "./usr/bin/google_guest_agent"
+    permissions="${arr[0]}"
+    deb_file_path="${arr[5]}"
+
+    # If the file doesn't have the "drwxr" permissions, is not located under '/usr/share' and the file is not "gce-workload-cert-refresh"...
+    if ! [[ "$permissions" == *"drwxr"* ]] && ! [[ "$deb_file_path" == *"/usr/share"* ]] && ! [[ "$deb_file_path" == *"gce-workload-cert-refresh"* ]]; then
+
+      # Write the deb_file_path to a replacement list file.
+      echo $deb_file_path >> $repl_file
+
+    fi
+  done <$file
+  cat $repl_file
+}
+
+validate_env() {
+  local error=false
+  local v
+
+  for v in "${REQUIRED_ENV_VARS[@]}"; do
```

Comment:
who's defining this `REQUIRED_ENV_VARS`?

Reply:
Function was outdated- removed.